inc/compat/plugins/compat-plugin-mailpoet.php

```diff
@@ -0,0 +1,137 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Compatibility with plugin: MailPoet integration.
+ */
+class FluidCheckout_MailPoet extends FluidCheckout {
+
+    /**
+     * __construct function.
+     */
+    public function __construct( ) {
+        add_action( 'init', [ $this, 'hooks' ] );
+    }
+
+    /**
+     * Initialize hooks.
+     */
+    public function hooks() {
+        // Bail when no mail poet class
+        if ( ! class_exists( MailPoet\Config\HooksWooCommerce::class ) ) {
+            return;
+        }
+
+        $this->remove_class_action( 'woocommerce_checkout_before_terms_and_conditions', 'MailPoet\Config\HooksWooCommerce',  'extendWooCommerceCheckoutForm' );
+        add_action( 'fc_checkout_contact_after_fields', [ $this, 'extend_woocommerce_checkout_form' ], 10 );
+    }
+
+    /**
+     * Extend WooCommerce checkout form for checkbox
+     *
+     * @return html
+     */
+    public function extend_woocommerce_checkout_form() {
```

Comment:
@alamgircsebd the original function `extendWooCommerceCheckoutForm` is public.

Why do we need to replace it with a modified version?

Reply:
Here I tried to call their original function, but it not working, because the class construct has some parameters. Without those parameters can't use the class.

Reply:
@alamgircsebd maybe you can use the function `get_object_by_class_name_from_hooks` to get the instance of the `HooksWooCommerce` class and use the functions from the already instantiated object.

Reply:
I used `get_object_by_class_name_from_hooks` but still it not working